src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1413,39 +1413,24 @@ public override BoundNode VisitAnonymousObjectCreationExpression(BoundAnonymousO
         {
             Debug.Assert(!IsConditionalState);
 
-            int receiverSlot = -1;
+            var anonymousType = (NamedTypeSymbol)node.Type;
+            Debug.Assert(anonymousType.IsAnonymousType);
+
             var arguments = node.Arguments;
-            var constructor = node.Constructor;
-            for (int i = 0; i < arguments.Length; i++)
+            var argumentTypes = arguments.SelectAsArray(arg => VisitRvalueWithResult(arg));
```

Comment:
>SelectAsArray [](start = 42, length = 13)

Consider either manually looping, or using the overload with an extra arg to pass the walker so the lambda can be cached.

Reply:
Good catch, thanks.

---
In reply to: [259068196](https://github.com/dotnet/roslyn/pull/33562#discussion_r259068196) [](ancestors = 259068196)